Failed to build R package.

After #5871 I can't build R package on my local machine with link time error:

```
/usr/bin/ld: ../../R-package/CMakeFiles/xgboost-r.dir/src/init.c.o: in function `R_init_xgboost':
/home/fis/Workspace/XGBoost/xgboost/R-package/src/init.c:83: undefined reference to `R_registerRoutines'
/usr/bin/ld: /home/fis/Workspace/XGBoost/xgboost/R-package/src/init.c:84: undefined reference to `R_useDynamicSymbols'
/usr/bin/ld: ../../R-package/CMakeFiles/xgboost-r.dir/src/xgboost_R.cc.o: in function `XGCheckNullPtr_R':
/home/fis/Workspace/XGBoost/xgboost/R-package/src/xgboost_R.cc:41: undefined reference to `R_ExternalPtrAddr'
/usr/bin/ld: /home/fis/Workspace/XGBoost/xgboost/R-package/src/xgboost_R.cc:41: undefined reference to `Rf_ScalarLogical'
/usr/bin/ld: ../../R-package/CMakeFiles/xgboost-r.dir/src/xgboost_R.cc.o: in function `_DMatrixFinalizer(SEXPREC*)':

...
```

R: 3.6.1
CMake: 3.13.4
GCC: gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008

--- a/R-package/CMakeLists.txt
+++ b/R-package/CMakeLists.txt
@@ -21,7 +21,7 @@ target_include_directories(xgboost-r
   ${PROJECT_SOURCE_DIR}/include
   ${PROJECT_SOURCE_DIR}/dmlc-core/include
   ${PROJECT_SOURCE_DIR}/rabit/include)
-target_link_libraries(xgboost-r PRIVATE ${LIBR_CORE_LIBRARY})
+target_link_libraries(xgboost-r PUBLIC ${LIBR_CORE_LIBRARY})
 set_target_properties(
   xgboost-r PROPERTIES
   CXX_STANDARD 14

--- a/R-package/CMakeLists.txt
+++ b/R-package/CMakeLists.txt
@@ -22,6 +22,10 @@ target_include_directories(xgboost-r
   ${PROJECT_SOURCE_DIR}/dmlc-core/include
   ${PROJECT_SOURCE_DIR}/rabit/include)
 target_link_libraries(xgboost-r PUBLIC ${LIBR_CORE_LIBRARY})
+if (USE_OPENMP)
+  find_package(OpenMP REQUIRED)
+  target_link_libraries(xgboost-r PUBLIC OpenMP::OpenMP_CXX OpenMP::OpenMP_C)
+endif (USE_OPENMP)
 set_target_properties(
   xgboost-r PROPERTIES
   CXX_STANDARD 14

